File: java/socketd/src/main/java/org/noear/socketd/transport/core/stream/StreamRequest.java
package org.noear.socketd.transport.core.stream;

import org.noear.socketd.transport.core.*;

import java.util.concurrent.CompletableFuture;

/**
 * 请求流
 *
 * @author noear
 * @since 2.0
 */
public class StreamRequest extends StreamBase {
    private final CompletableFuture<Reply> future;

    public StreamRequest(String sid, long timeout, CompletableFuture<Reply> future) {
        super(sid, true, timeout);
        this.future = future;
    }

    /**
     * 是否结束接收
     */
    @Override
    public boolean isDone() {
        return future.isDone();
    }

    /**
     * 接收时
     */
    @Override
    public void onAccept(MessageInternal reply, Channel channel) {
        future.complete(reply);
    }
}

File: dotnet/socketd/org.noear.socketd/transport/core/stream/StreamRequest.cs
namespace org.noear.socketd.transport.core;

public class StreamRequest : StreamBase {
    private Action<IReply> future;

    public StreamRequest(String sid, long timeout, Action<IReply> future) : base(sid, true, timeout) {
        this.future = future;
    }

    /**
     * 是否结束接收
     */
    public override bool isDone() {
        return false;
    }

    /**
     * 接收时
     */

    public override void onAccept(IMessageInternal reply, IChannel channel) {
        future.Invoke(reply);
    }
}